bundles/io/org.eclipse.smarthome.io.rest.core/src/main/java/org/eclipse/smarthome/io/rest/core/persistence/PersistenceResource.java

```diff
@@ -69,68 +81,129 @@
     // The URI path to this resource
     public static final String PATH = "persistence";
 
-    static private Map<String, PersistenceService> persistenceServices = new HashMap<String, PersistenceService>();
+    private ItemRegistry itemRegistry;
+    private Map<String, PersistenceService> persistenceServices = new HashMap<String, PersistenceService>();
 
     public void addPersistenceService(PersistenceService service) {
-        persistenceServices.put(service.getName(), service);
+        persistenceServices.put(service.getId(), service);
     }
 
     public void removePersistenceService(PersistenceService service) {
-        persistenceServices.remove(service.getName());
+        persistenceServices.remove(service.getId());
+    }
+
+    protected void setItemRegistry(ItemRegistry itemRegistry) {
+        this.itemRegistry = itemRegistry;
+    }
+
+    protected void unsetItemRegistry(ItemRegistry itemRegistry) {
+        this.itemRegistry = null;
     }
 
     @GET
```

Comment:
As PersistenceServices now have a localizeable label, you should accept a locale as an optional parameter as well (see e.g. [here](https://github.com/eclipse/smarthome/blob/e0985e90d389350580e59bfbcd7e0adaaccf7519/bundles/io/org.eclipse.smarthome.io.rest.core/src/main/java/org/eclipse/smarthome/io/rest/core/thing/ThingTypeResource.java#L110))